diff --git a/dlls/ntdll/unix/sync.c b/dlls/ntdll/unix/sync.c
index 67ab945c..07f654e4 100644
--- a/dlls/ntdll/unix/sync.c
+++ b/dlls/ntdll/unix/sync.c
@@ -2923,10 +2923,14 @@ NTSTATUS WINAPI NtRemoveIoCompletionEx( HANDLE handle, FILE_IO_COMPLETION_INFORM
 
     if (!count) return STATUS_INVALID_PARAMETER;
 
-    if (timeout && !timeout->QuadPart && inproc_device_fd >= 0)
+    if (timeout && !timeout->QuadPart && do_fsync())
     {
         status = NtWaitForSingleObject( handle, alertable, timeout );
-        if (status != WAIT_OBJECT_0) goto done;
+        if (status != STATUS_SUCCESS)
+        {
+            if (written) *written = 0;
+            return status;
+        }
     }
 
     while (i < count)
@@ -2977,7 +2981,7 @@ NTSTATUS WINAPI NtRemoveIoCompletionEx( HANDLE handle, FILE_IO_COMPLETION_INFORM
     SERVER_END_REQ;
 
 done:
-    *written = i ? i : 1;
+    if (written) *written = i;
     return status;
 }
 
